Analysis/WWCategorisation/src/WWCategorisationProcessor.cc

```diff
@@ -0,0 +1,436 @@
+#include "WWCategorisationProcessor.h"
+
+#include <iostream>
+#include <sstream>
+#include <cmath>
+#include <vector>
+#include <array>
+#include <algorithm>
+
+#include "EVENT/LCCollection.h"
+#include "EVENT/ReconstructedParticle.h"
+#include "IMPL/LCEventImpl.h" 
+#include "IMPL/LCCollectionVec.h"
+#include "IMPL/MCParticleImpl.h" 
+#include "IMPL/ReconstructedParticleImpl.h"
+
+#include "TText.h"
+#include "TStyle.h"
+#include "TLorentzVector.h"
+
+using namespace lcio;
+using namespace marlin;
+
+
+WWCategorisationProcessor  aWWCategorisationProcessor ;
+
+WWCategorisationProcessor::WWCategorisationProcessor() : Processor("WWCategorisationProcessor") {
+  
+  _description = "Categorisation of all WW decay channels" ;
+  
+  registerInputCollection(LCIO::MCPARTICLE,
+			   "MCParticles",
+			   "Name of the MCParticle collection",
+			   _MCParColName,
+			   std::string("MCParticlesSkimmed"));
+  registerInputCollection(LCIO::RECONSTRUCTEDPARTICLE,
+         "IsolatedElectrons",
+         "Name of the ReconstructedParticle collection",
+         _ElectronColName,
+         std::string("IsolatedElectrons"));
+  registerInputCollection(LCIO::RECONSTRUCTEDPARTICLE,
+			   "IsolatedMuons",
+			   "Name of the ReconstructedParticle collection",
+			   _MuonColName,
+			   std::string("IsolatedMuons"));
+  registerInputCollection(LCIO::RECONSTRUCTEDPARTICLE,
+			   "IsolatedTaus",
+			   "Name of the ReconstructedParticle collection",
+			   _TauColName,
+			   std::string("IsolatedTaus"));
+  registerInputCollection( LCIO::RECONSTRUCTEDPARTICLE,
+			   "IsolatedPhotons",
+			   "Name of the ReconstructedParticle collection",
+			   _PhotonColName,
+			   std::string("IsolatedPhotons"));
+  registerInputCollection(LCIO::RECONSTRUCTEDPARTICLE,
+			   "PFOsminusphoton",
+			   "Name of the ReconstructedParticle collection",
+			   _PFOsMinusIsolatedObjetcs,
+			   std::string("PFOsminusphoton"));
+
+  registerProcessorParameter("TTreeFileName",
+          "Name of the root file in which the TTree with the 4 event observables is stored; if left empty no root file is created; default: TTreeFile.root",
+          _TTreeFileName,
+          std::string("TTreeFile.root"));
+  registerProcessorParameter("ConfusionMatrixFileName",
+          "Name of the png file in which the confusion matrix is stored; if left empty no file is created; default: ConfusionMatrix.png",
+          _ConfusionMatrixFileName,
+          std::string("ConfusionMatrix.png"));
+}
```

Comment:
I know it is a bit pedantic but can you indent this consistently?

Reply:
Oof, I didn't even realise that is there... my text editor shows the tabs as 2 spaces and then it fits.
Replaced the tabs with spaces, should be ok now.